server_config.cc

```diff
@@ -41,83 +34,156 @@ std::string buildWordyParam(std::shared_ptr<NginxConfigStatement> statement,
   return paramName;
 }
 
+// ServerConfig acts as a wrapper for NginxConfig objects, allowing for easier
+// retrieval of things like port, child blocks, and mappings from route
+// prefixes to handler names
+ServerConfig::ServerConfig(NginxConfig config)
+  : config_(config)
+{
+  // Store raw NginxConfig for use in getting child blocks
+}
+
+// Initializer for a ServerConfig object, Build() iterates through the passed
+// in config, creating both the map from path to property values as well as
+// the map from prefix to handler_name.
+bool ServerConfig::Build() {
+  std::vector<std::string> basePath = {};
+  printPropertiesMap();
+  return fillOutMaps(config_, basePath);
+}
+
 // In NgnixConfig, Statements are root level 'blocks' (e.g. server {})
-// fillOutMap will recursively construct an unordered_map between a vector of strings
+// fillOutMaps will recursively construct an unordered_map between a vector of strings
 // representing the 'path' to a config parameter and the value of that parameter
 //   e.g. ["server", "listen"] should be mapped to the port number
-void ServerConfig::fillOutMap(NginxConfig config, std::vector<std::string> basePath) {
+bool ServerConfig::fillOutMaps(NginxConfig config, std::vector<std::string> basePath) {
   if (config.statements_.size() < 1) {
-    // TODO: Possible logging here about empty config
-    return;
+    std::cout << "Failed to build ServerConfig map, recieved empty config\n";
+    return false;
   }
 
   // Loop through all config param block at the current depth
   for (size_t i = 0; i < config.statements_.size(); i++) {
-    size_t numTokens = config.statements_[i]->tokens_.size();
+    std::shared_ptr<NginxConfigStatement> curStatement = config.statements_[i];
```

Comment:
Like Ky-Cuong said, `snake_case` for variables, so `curStatement` should be changed to `cur_statement`?